.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,27 @@
+name: pre-commit
+
+on:
+  pull_request:
+
+permissions:
+  contents: read
+  pull-requests: read
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-python@v3
+    - uses: pre-commit/action@646c83fcd040023954eafda54b4db0192ce70507
+      with: ## let's skip to run for all the files
+        extra_args: --help
+    - id: files
+      uses: jitterbit/get-changed-files@b17fbb00bdc0c0f63fcf166580804b4d2cdc2a42
+    - name: Configure PATH
+      run: echo "${GITHUB_WORKSPACE}/.ci/scripts" >> $GITHUB_PATH
+    - name: Precommit changes
+      run: |
+        for changed_file in ${{ steps.files.outputs.all }}; do
+          pre-commit run --files "${changed_file}"
+        done
```

Comment:
I could not find a better way to run `pre-commit` for the subset of changes.

Reply:
Honestly I'd be fine with running `pre-commit` on everything. It should be very fast. But this also seems fine 🤷‍♂️

Reply:
Wouldn't this also work without the for loop?

    pre-commit run --files ${{ steps.files.outputs.all }}

[`pre-commit run --files`](https://pre-commit.com/#pre-commit-run) can take a list of files. Not sure if we could run into some limit if the change affects hundreds of files, though (e.g. when Python 4.0 moves from spaces to tabs ;) )

Reply:
good point, I'll try it in a bit :)